modules/data-entry/src/main/frontend/src/resourceQuery/ResourceQuery.jsx

```diff
@@ -0,0 +1,525 @@
+              closeSuggestions();
+            }}
+          >
+            <div>
+            { suggestion.label }
+            { infoDisplayer &&
+              <IconButton
+                size="small"
+                buttonRef={node => {
+                  registerInfoButton(suggestion["@path"], node);
+                }}
+                color="primary"
+                onClick={(e) => {setResourcePath(element["@path"]); e.preventDefault(); e.stopPropagation();}}
+                className={classes.infoButton}
+            >
+              <Info color="primary" />
+            </IconButton>
+            }
+            { suggestion.matchedFields?.map(f =>
+              <FormattedText
+                key={f}
+                component="div"
+                variant="caption"
+                color="textSecondary"
+              >
+                {f}
+              </FormattedText>
+            )}
+            </div>
+          </MenuItem>
+        );
+      });
+    }
+
+    if (showUserEntry) {
+      suggestions.length > 0 && suggestions.push(<Divider key="divider"/>);
+      suggestions.push(
+        <MenuItem
+          className={classes.dropdownItem}
+          key={NO_RESULTS_TEXT}
+          disabled={true}
+        >
+          <Typography
+            component="p"
+            className={classes.noResults}
+            variant="caption"
+          >
+            {data["rows"].filter(r => !r.error).length > 0 ? NONE_OF_ABOVE_TEXT : NO_RESULTS_TEXT}
+          </Typography>
+        </MenuItem>
+      );
+
+      suggestions.push(
+        <MenuItem
+          className={classes.dropdownItem}
+          key={anchorEl.current.value}
+          onClick={(e) => {
+              if (e.target.localName === "li") {
+                onClick(anchorEl.current.value, anchorEl.current.value);
+                clearOnClick && setInputValue("");
+                closeSuggestions();
+              }}
+            }
+        >
+          {anchorEl.current.value}
+        </MenuItem>
+      );
+    }
+
+    setSuggestions(suggestions);
+    setSuggestionsVisible(true);
+  }
+
+  // Event handler for clicking away from the autocomplete while it is open
+  let closeAutocomplete = event => {
+    if ( browserRef?.current
+      || menuPopperRef?.current?.contains(event.target)
+      || infoboxRef?.current?.contains(event.target)) {
+      return;
+    }
+
+    !anchorEl?.current?.contains(event.target)
+      && !searchButtonRef?.current?.contains(event.target)
+      && maxAnswers !== 1
+      && clearOnClick
+      && setInputValue("");
+    setSuggestionsVisible(false);
+    setResourcePath("");
+  };
+
+  let InfoDisplayer = infoDisplayer;
+
+    // Register a button reference that the info box can use to align itself to
+  let registerInfoButton = (id, node) => {
+    // List items getting deleted will overwrite new browser button refs, so
+    // we must ignore deregistration events
+    if (node) {
+      buttonRefs[id] = node;
```

Comment:
Prefer to always use the setter when modifying a state object, as this will push a rerender onto whatever element is being used as `InfoDisplayer` when the ref changes (and it might need to be notified when it does):
```suggestion
      buttonRefs[id] = node;
```

Reply:
Addressed in [bf9b90e](https://github.com/data-team-uhn/cards/pull/1043/commits/bf9b90e18010034433e18f69faa7e4537ecad1d6) on the `CARDS-1784` branch.